Orange/widgets/visualize/owscatterplot.py

```diff
@@ -380,10 +380,16 @@ def update_graph(self, attributes=None, reset_view=True, **_):
         self.graph.zoomStack = []
         if attributes and len(attributes) == 2:
             self.attr_x, self.attr_y = attributes
+        X = BasicStats(self.data[:, self.attr_x], self.attr_x)
+        Y = BasicStats(self.data[:, self.attr_y], self.attr_y)
```

Comment:
I have not really looked in depth at this problem, so I can't say off the top of my head. But I think it is good exercise for you to try and work out a solution yourself.